debian/tracker/debian.go

```diff
@@ -158,8 +159,26 @@ func (c Client) update(dirname string, bugs []Bug) error {
 	for _, bug := range bugs {
 		dir := filepath.Join(c.vulnListDir, trackerDir, dirname)
 		if dirname == "CVE" {
-			if err := utils.SaveCVEPerYear(dir, bug.Header.ID, bug); err != nil {
-				return xerrors.Errorf("debian: failed to save CVE per year: %w", err)
+			if strings.HasSuffix(bug.Header.ID, "-XXXX") {
+				var bugno int
+				for _, ann := range bug.Annotations {
+					if ann.Type == "package" && ann.BugNo != 0 {
+						bugno = ann.BugNo
+						break
+					}
+				}
+
+				bug.Header.ID = tempBugName(bugno, bug.Header.Description)
+
+				fileName := fmt.Sprintf("%s.json", bug.Header.ID)
+				filePath := filepath.Join(dir, "TEMP", fileName)
+				if err := utils.Write(filePath, bug); err != nil {
+					return xerrors.Errorf("debian: write error (%s): %w", filePath, err)
+				}
```

Comment:
Maybe we want to move this to separate function?

Reply:
The logic as a whole is not that complicated, and the part that generates a new ID using the bug number and description has already been extracted into a function, so I don't think any more is needed.